sql-to-dbsp-compiler/lib/sqllib/src/lib.rs

```diff
@@ -18,6 +19,291 @@ use std::ops::Add;
+            }
+
+            pub fn [<$func_name NN_>]( arg0: Option<$arg_type0>, arg1: Option<$arg_type1>, arg2: $arg_type2 ) -> Option<$ret_type> {
+                let arg0 = arg0?;
+                let arg1 = arg1?;
+                Some([<$func_name ___>](arg0, arg1, arg2))
+            }
+
+            pub fn [<$func_name NNN>]( arg0: Option<$arg_type0>, arg1: Option<$arg_type1>, arg2: Option<$arg_type2> ) -> Option<$ret_type> {
+                let arg0 = arg0?;
+                let arg1 = arg1?;
+                let arg2 = arg2?;
+                Some([<$func_name ___>](arg0, arg1, arg2))
+            }
+        }
+    }
+}
+
+// Macro to create variants of a function with 2 arguments
+// If there exists a function is f_t_t(x: T, y: T) -> U, this creates
+// three functions:
+// - f_tN_t(x: T, y: Option<T>) -> Option<U>
+// - f_t_tN(x: Option<T>, y: T) -> Option<U>
+// - f_tN_tN(x: Option<T>, y: Option<T>) -> Option<U>
+// The resulting functions return Some only if all arguments are 'Some'.
+#[macro_export]
+macro_rules! some_operator {
+    ($func_name: ident, $short_name: ident, $arg_type: ty, $ret_type: ty) => {
+        ::paste::paste! {
+            #[inline(always)]
+            pub fn [<$func_name _ $short_name N _ $short_name N>]( arg0: Option<$arg_type>, arg1: Option<$arg_type> ) -> Option<$ret_type> {
+                let arg0 = arg0?;
+                let arg1 = arg1?;
+                Some([<$func_name _$short_name _ $short_name>](arg0, arg1))
+            }
+
+            #[inline(always)]
+            pub fn [<$func_name _ $short_name _ $short_name N>]( arg0: $arg_type, arg1: Option<$arg_type> ) -> Option<$ret_type> {
+                let arg1 = arg1?;
+                Some([<$func_name _ $short_name _ $short_name>](arg0, arg1))
+            }
+
+            #[inline(always)]
+            pub fn [<$func_name _ $short_name N _ $short_name>]( arg0: Option<$arg_type>, arg1: $arg_type ) -> Option<$ret_type> {
+                let arg0 = arg0?;
+                Some([<$func_name _ $short_name _ $short_name>](arg0, arg1))
+            }
+        }
+    }
+}
+
+#[macro_export]
+macro_rules! for_all_int_compare {
+    ($func_name: ident, $ret_type: ty) => {
+        some_operator!($func_name, i16, i16, bool);
+        some_operator!($func_name, i32, i32, bool);
+        some_operator!($func_name, i64, i64, bool);
+    }
+}
+
+#[macro_export]
+macro_rules! for_all_numeric_compare {
+    ($func_name: ident, $ret_type: ty) => {
+        for_all_int_compare!($func_name, bool);
+        some_operator!($func_name, f, F32, bool);
+        some_operator!($func_name, d, F64, bool);
+    }
+}
+
+#[macro_export]
+macro_rules! for_all_compare {
+    ($func_name: ident, $ret_type: ty) => {
+        for_all_numeric_compare!($func_name, bool);
+        some_operator!($func_name, b, bool, bool);
+        some_operator!($func_name, s, String, bool);
+    }
+}
+
+#[macro_export]
+macro_rules! for_all_int {
```

Comment:
The previous macros make it clear they are for comparison operators. Maybe name these `for_all_int_arithmetic` and `for_all_numeric_arithmetic`?